cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -105,6 +111,28 @@ func SetDefaults_MasterConfiguration(obj *MasterConfiguration) {
 
 	SetDefaultsEtcdSelfHosted(obj)
 	SetDefaults_KubeletConfiguration(obj)
+	SetDefaults_ProxyConfiguration(obj)
+
+}
+
+// SetDefaults_ProxyConfiguration asigns default values for the Proxy
+func SetDefaults_ProxyConfiguration(obj *MasterConfiguration) {
```

Comment:
as the proxy config should be a pointer, you should do the same as the kubelet config below
```go
if obj.KubeletConfiguration.BaseConfig == nil {
     obj.KubeletConfiguration.BaseConfig = &kubeletconfigv1alpha1.KubeletConfiguration{}
}
```
i.e.
```go
if obj.KubeProxy.Config == nil {
     obj.KubeProxy.Config = &kubeproxyconfigv1alpha1.KubeProxyConfiguration{}
}
```

Reply:
done

Reply:
@luxas Does this function, via inspection, end up setting defaults for all the nested fields? Does it also so version conversion?  I'm wondering because, when I use this patch-set, I get an error where it says it cannot convert the ClientConnection.Burst field, indicating that it is int32 in baseline and int in v1alpha1.

Wondering how that is handled.

Reply:
I replied below that defaulting for the rest of the config should be set
The external type should always be used when serializing/deserializing, the internal type might be needed for internal purposes you might have stumbled upon